R/linearregutils.R

```diff
@@ -141,3 +159,51 @@ createPlotOutputsLinearOSR <- function(the.model){
 createPlotOutputsLinearXDF <- function(){
   noDiagnosticPlot("The diagnostic plot is not available for XDF based models")
 }
+
+#' Prepare the glmnet/cv.glmnet plot featuring the L1 norm on the x axis.
+#'
+#' @param the.model model object
+#' @export
+createFirstPlotOutputGLMNET <- function(the.model){
+  if (inherits(the.model, "cv.glmnet")) {
+    plotted_obj <- the.model$glmnet.fit
+  } else {
+    plotted_obj <- the.model
+  }
+  plot(the.model, xvar = "norm")
+}
+
+#' Prepare the glmnet/cv.glmnet plot featuring log(lambda) on the x axis.
+#'
+#' @param the.model model object
+#' @export
+createSecondPlotOutputGLMNET <- function(the.model){
+  if (inherits(the.model, "cv.glmnet")) {
+    plotted_obj <- the.model$glmnet.fit
+  } else {
+    plotted_obj <- the.model
+  }
+  plot(the.model, xvar = "lambda")
+}
+
+#' Prepare the glmnet/cv.glmnet plot featuring the fraction of deviance
+#' explained on the x axis.
+#'
+#' @param the.model model object
+#' @export
+createThirdPlotOutputGLMNET <- function(the.model){
+  if (inherits(the.model, "cv.glmnet")) {
+    plotted_obj <- the.model$glmnet.fit
+  } else {
+    plotted_obj <- the.model
+  }
+  plot(the.model, xvar = "dev")
+}
+
```

Comment:
This is a perfect example of where you want to use the `DRY` principle

```r
createPlotOutputsGLMNET <- function(the.model, xvar, ...){
  plot_obj <- if (inherits(the.model, "cv.glmnet")) the.model$glmnet.fit else the.model
  function(){plot(the.model, xvar = xvar)}
}
```

You can now use it as

```r
xvars <- c('norm', 'lambda', 'dev')
glmPlotOutputs <- plyr::llply(xvars, createPlotOutputsGLMNET)
```